tests/unit/CMakeLists.txt

```diff
@@ -31,14 +31,15 @@ endif()
 include(${root}/cmake/x86_64InstructionFlags.cmake)
 
 add_executable(test_hnsw ../utils/mock_thread_pool.cpp test_hnsw.cpp test_hnsw_multi.cpp test_hnsw_tiered.cpp unit_test_utils.cpp)
-add_executable(test_hnsw_parallel test_hnsw_parallel.cpp unit_test_utils.cpp)
-add_executable(test_bruteforce test_bruteforce.cpp test_bruteforce_multi.cpp unit_test_utils.cpp)
-add_executable(test_allocator test_allocator.cpp unit_test_utils.cpp)
+add_executable(test_hnsw_parallel test_hnsw_parallel.cpp ../utils/mock_thread_pool.cpp unit_test_utils.cpp)
+add_executable(test_bruteforce test_bruteforce.cpp test_bruteforce_multi.cpp ../utils/mock_thread_pool.cpp unit_test_utils.cpp)
+add_executable(test_allocator test_allocator.cpp ../utils/mock_thread_pool.cpp unit_test_utils.cpp)
```

Comment:
I `CreateTieredParams` a function to `unit_test_utils.cpp` that uses `tieredIndexMock::submit_callback`, defined in `mock_thread_pool.cpp`, so `mock_thread_pool.cpp` and `unit_test_utils.cpp` should always be compiled together to avoid linker errors.

Reply:
I see. I still suggest separating tiered and its deps, maybe having a "tiered_utils" file and link to that where necessary

Reply:
Yeah we should do it and cherry pick it to avoid dealing with conflicts during every backport.
I've already created a ticket to move all test utils into a dedicated namespace. I'll expand the scope of that task to include moving all tiered utils into a unified file as well.

Reply:
MOD-8345